components/add_new_cv_form.tsx

```diff
@@ -127,34 +128,70 @@ export default function AddNewCvForm({ id }: Props) {
       return await supabase.from("certifications").delete().eq("cv_id", cvId);
     }
 
+    if (certificationsToRemove.length !== 0) {
+      await supabase
+        .from("certifications")
+        .delete()
+        .in("id", certificationsToRemove);
+    }
+
     const updatedCertifications = certifications.map((certification: any) => ({
       ...certification,
       cv_id: cvId,
+      id: certification.id || null,
+      created_at: certification.created_at || null,
     }));
     return supabase.from("certifications").upsert(updatedCertifications);
   }
 
-  async function upsertEducation(education: any, cvId: string) {
-    education.cv_id = cvId;
-    return supabase.from("education").upsert(education);
+  async function upsertEducation(educations: any, cvId: string) {
+    if (educations.length === 0) {
+      return await supabase.from("educations").delete().match({ cv_id: cvId });
+    }
+
+    if (educationsToRemove.length !== 0) {
+      await supabase.from("educations").delete().in("id", educationsToRemove);
+    }
+
+    const updatedEducations = educations.map((education: any) => {
+      return {
+        ...education,
+        cv_id: cvId,
+        id: education.id || null,
+        created_at: education.created_at || null,
+      };
+    });
+
+    return supabase.from("educations").upsert(updatedEducations);
   }
 
   async function upsertProjects(projects: any, cvId: string) {
     if (projects.length === 0) {
       return await supabase.from("projects").delete().eq("cv_id", cvId);
     }
 
+    if (projectsToRemove.length !== 0) {
+      await supabase.from("projects").delete().in("id", projectsToRemove);
+    }
+
     const updatedProjects = projects.map((project: any) => {
       const startDate = new Date(project.date_start);
       startDate.setDate(15);
 
-      const endDate = new Date(project.date_end);
-      endDate.setDate(15);
+      let endDate = null;
+      if (project.date_end) {
+        endDate = new Date(project.date_end);
+        endDate.setDate(15);
+      }
       return {
         ...project,
         cv_id: cvId,
         date_start: startDate.toISOString(),
-        date_end: endDate.toISOString(),
+        date_end: endDate?.toISOString() || null,
+        id: project.id || null,
```

Comment:
If we pass undefined instead of null, we get an error `All object keys must match`. I think that's because we made a trigger in supabase to check if the value is `NULL` and if it is, create a new `ID`.